checks.d/agent_metrics.py

```diff
@@ -113,6 +114,11 @@ def get_metric_context(self):
         return self._collector_payload, self._metric_context
 
     def check(self, instance):
+
+        if Platform.is_linux():
+            procfs_path = self.agentConfig.get('procfs_path', '/proc').rstrip('/')
+            psutil.PROCFS_PATH = procfs_path
```

Comment:
Would it be possible to move this logic into the definition of the AgentCheck class? I'm pretty sure someone will forget to add this block in a new check at some point and we'll introduce bugs that could be easily avoided if this is called earlier and automatically.

The danger would remain for calls to `/proc` subfolders from `utils/`, `agent.py` and `config.py` but we don't modify these files as often and we are far more careful when doing so.

Let me know what you think.